### PR TITLE
Ethernet and IP addresses pattern tests

### PR DESCRIPTION
**Pull request by Rangababu-R**
Added Test Cases for Ethernet addresses
and IP address pattern validation.
covers: Issues #104  and #87

### CODE REVIEW

**Comment by ashutshkumr** [comment]
@Rangababu-R can you please check if python2 failur is legimitate on your setup.

File: tests/ethernet/eth.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import utils
```
Comment (ashutshkumr) [inline tests/ethernet/eth.py:1]: Please move all common functions to utils itself. If you think it belongs to a specific separate module, feel free to create eth.py inside `utils/`

File: tests/ethernet/eth.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def captures_ok(api, cfg, size, src, dst):
```
Comment (ashutshkumr) [inline tests/ethernet/eth.py:4]: after moving it to utils, see if we can call it like so:
```python
utils.eth.captures_ok()
```

It's not impossible, but if it's too much effort for now, consider renaming it to add prefix.

File: tests/ethernet/eth.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    Returns normally if patterns in captured packets are as expected.
+    """
+    cap_dict = utils.get_all_captures(api, cfg)
+    src_list = []
+    dst_list = []
+    length = []
+    for k in cap_dict:
+        for b in cap_dict[k]:
+            dst_list.append(
+                ":".join(['{:02X}'.format(byt) for byt in b[0:6]])
```
Comment (ashutshkumr) [inline tests/ethernet/eth.py:15]: wouldn't it make sense to move it to `mac_addr_from_bytes(b[0:6])` ? If we haven something similar for IPv4 as well, we should consider the same for it as well.

File: tests/ethernet/test_counter_mac_addr.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+from abstract_open_traffic_generator import flow
+from abstract_open_traffic_generator.flow import Counter
```
Comment (ashutshkumr) [inline tests/ethernet/test_counter_mac_addr.py:2]: let's be consistent here by simply using `flow.Counter` like all other flow components.

File: tests/ipv4/ipv4.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import utils
```
Comment (ashutshkumr) [inline tests/ipv4/ipv4.py:1]: Requesting approach similar to discussed in mac addr section.

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
As I didn't get a chance to review yesterday the `generate*` functions. Here's what I suggest:
- All function name should follow consistent naming based on what they return and what pattern the accept. Something like `get_mac_addrs_from_counter_pattern()` or `mac_addrs_from_counter_pattern()`. I see some names which mixed counter with list (not intentional though).
- Reduce sample size of examples for `generate*` functions from 10 to 5 or 3. It's sufficient to showcase intent.
- Add functions `mac_addr_to_num()` and `num_to_mac_addr()`. Similarly for IPv4 addr. The use of `ipaddr` library isn't quite justified IMHO. The `generate` functions should be using those functions, resulting in a more reasonable code-block.
- We refactored `stats_ok` out of tests. Why not just refactor `wait_for` and call it `wait_for_stats_ok` ?

File: tests/ethernet/eth.py (hunk 3 above)
Comment (Rangababu-R) [reply tests/ethernet/eth.py:15]: As this was added at protocol level, did not feel to add as definition.
will move to definition

**Comment by Rangababu-R** [comment]
> 
> 
> As I didn't get a chance to review yesterday the `generate*` functions. Here's what I suggest:
> 
>     * All function name should follow consistent naming based on what they return and what pattern the accept. Something like `get_mac_addrs_from_counter_pattern()` or `mac_addrs_from_counter_pattern()`. I see some names which mixed counter with list (not intentional though).
> 
>     * Reduce sample size of examples for `generate*` functions from 10 to 5 or 3. It's sufficient to showcase intent.
> 
>     * Add functions `mac_addr_to_num()` and `num_to_mac_addr()`. Similarly for IPv4 addr. The use of `ipaddr` library isn't quite justified IMHO. The `generate` functions should be using those functions, resulting in a more reasonable code-block.
> 
>     * We refactored `stats_ok` out of tests. Why not just refactor `wait_for` and call it `wait_for_stats_ok` ?

Done changes with restpy and removed ipaddr lib and made counter pattern local

**Pull request merged**